utils.R

```diff
@@ -0,0 +1,57 @@
+#' GetMissingColumns
+#' 
+#' Returns a character vector of missing required columns
+#' from a given dataframe
+#' 
+#' Required columns are:
+#' author, year, TP, FN, FP, TN
+#'
+#' @param df a dataframe
+#'
+#' @return a character vector of missing columns 
+#' (NULL if none are missing)
+GetMissingCols <- function(df) {
```

Comment:
### Optional
This function could be simplified to:
```
required_cols <- c("author", "year", "TP", "FN", "FP", "TN")
return(required_cols [!required_cols %in% names(df)]
```
That might be less readable though.